pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -393,25 +393,20 @@ func validateActions(ctx context.Context, log *logrus.Entry, r *azure.Resource,
 		permissions, err := client.ListForResource(ctx, r.ResourceGroup, r.Provider, "", r.ResourceType, r.ResourceName)
 		if detailedErr, ok := err.(autorest.DetailedError); ok &&
 			detailedErr.StatusCode == http.StatusForbidden {
-			log.Print(err)
-			err = authorizer.RefreshWithContext(ctx)
-			if err != nil {
+			done, err := authorizer.RefreshWithContext(ctx, log)
+			if !done || err != nil {
 				return false, err
 			}
-			return false, nil
```

Comment:
Doesn't look quite right.  We can only be in this block if ListForResource returned Forbidden.  We are trying to refresh the token to have another go.  We must return false, nil or false, err.  I think this segment should be:

```golang
		if detailedErr, ok := err.(autorest.DetailedError); ok &&
			detailedErr.StatusCode == http.StatusForbidden {
			_, err = authorizer.RefreshWithContext(ctx, log)
			return false, err
		}
```

wdyt @nilsanderselde?